app/controllers/registration_commands_controller.rb

```diff
@@ -1,28 +1,59 @@
 class RegistrationCommandsController < ApplicationController
   include Foreman::Controller::RegistrationCommands
 
-  def new
-    form_options
+  def form_data
+    render json: {
+      organizations: organizations,
+      locations: locations,
+      hostGroups: Hostgroup.authorized(:view_hostgroups),
+      operatingSystems: Operatingsystem.authorized(:view_operatingsystems),
+      smartProxies: Feature.find_by(name: 'Registration')&.smart_proxies,
+      configParams: host_config_params,
+      pluginData: plugin_data,
+    }
+  end
+
+  def operatingsystem_template
+    os = Operatingsystem.authorized(:view_operatingsystems).find(params[:id])
+    template_kind = TemplateKind.find_by(name: 'host_init_config')
+    template = os.os_default_templates
+                 .find_by(template_kind: template_kind)&.provisioning_template
+
+    if template
+      render json: { template: { name: template.name, path: edit_provisioning_template_path(template) } }
+    else
+      render json: { template: { name: nil, os_path: edit_operatingsystem_path(os)} }
+    end
   end
 
   def create
-    form_options
-    @command = command
+    render json: { command: command }
   end
 
   private
 
-  def form_options
-    @host_groups = Hostgroup.authorized(:view_hostgroups).select(:id, :name)
-    @operating_systems = Operatingsystem.authorized(:view_operatingsystems).select(:id, :title)
-    @smart_proxies = Feature.find_by(name: 'Registration')&.smart_proxies || []
+  def organizations
+    User.current.admin? ? Organization.select(:id, :name) : User.current.my_organizations.select(:id, :name)
```

Comment:
It feels like `my_organizations` should take care of `User.current.admin?` so it could become:
```suggestion
    User.current.my_organizations.select(:id, :name)
```